MvvmCross/Windows/Uwp/Views/MvxWindowsViewPresenter.cs

```diff
@@ -28,34 +33,111 @@ public MvxWindowsViewPresenter(IMvxWindowsFrame rootFrame)
             SystemNavigationManager.GetForCurrentView().BackRequested += BackButtonOnBackRequested;
         }
 
-        private IMvxViewModelTypeFinder _viewModelTypeFinder;
-        public IMvxViewModelTypeFinder ViewModelTypeFinder
+        public override void RegisterAttributeTypes()
+        {
+            AttributeTypesToActionsDictionary.Add(
+                typeof(MvxPagePresentationAttribute),
+                new MvxPresentationAttributeAction
+                {
+                    ShowAction = (view, attribute, request) => ShowPage(request),
+                    CloseAction = (viewModel, attribute) => ClosePage(viewModel)
+                });
+
+            AttributeTypesToActionsDictionary.Add(
+                typeof(MvxSplitViewPresentationAttribute),
+                new MvxPresentationAttributeAction
+                {
+                    ShowAction = (view, attribute, request) => ShowSplitView((MvxSplitViewPresentationAttribute)attribute, request),
+                    CloseAction = (viewModel, attribute) => CloseSplitView(viewModel)
+                });
+        }
+
+        public override MvxBasePresentationAttribute CreatePresentationAttribute(Type viewModelType, Type viewType)
+        {
+            MvxTrace.Trace($"PresentationAttribute not found for {viewType.Name}. Assuming new page presentation");
+            return new MvxPagePresentationAttribute();
```

Comment:
Is there no way to detect a splitview?

Reply:
@martijn00 not without creating an instance of the viewType - not sure we should be encouraging that behaviour from developers as it'll only be negative for perf